libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1087,10 +1089,18 @@ bool TypeChecker::visit(VariableDeclarationStatement const& _statement)
 	{
 		// No initial value is only permitted for single variables with specified type.
 		if (_statement.declarations().size() != 1 || !_statement.declarations().front())
-			m_errorReporter.fatalTypeError(_statement.location(), "Assignment necessary for type detection.");
+		{
+			// Error already reported in syntax checker, but we need to bail out here?
```

Comment:
The error is not already reported in the syntax checker, is it? It's only if no variables are declared, isn't it? It's probably a good idea to bail out here, if there are in fact variables, since they may be referenced later and would then be expected to have a type annotation. But the comment here is a bit confusing.